cardano-node/Cardano/Node/Run.hs

```diff
@@ -178,8 +182,27 @@ handleSimpleNode p NodeCLIArguments{..}
                  nodeTraces
                  CardanoConfiguration{..}
     = do
+
+    -- Get filepath to genesis file
+    let genesisFilePath = cliGenesisFile commonCLI
+
+    let fp =
+          either
+            (\err -> error $ unpack err)
+            id
+            (lastToEither (pack "handleSimpleNode: No genesis filepath specified") genesisFilePath)
+
+    eitherFP <- runExceptT $ readGenesisData fp
```

Comment:
So reading the genesis file here isn't what we want. For two reasons:
1. we're in code that's polymorphic in the protocol, and not all protocols read a genesis file, only the "real" one and not the demo ones.
2. for the "real" case we've in fact already read the genesis file (see use of `Genesis.mkConfigFromFile` in `cardano-node/Cardano/Node/CLI.hs`).

So the genesis config is actually here already, but it's an argument to the `ProtocolRealPBFT` constructor for the `Consensus.Protocol blk` arg to this function. So yes this means it's only available for the one case, not all cases. Urk!

So I suggest you work with @mrBliss to resolve this by making access to the system start time be a generic thing. For example, it could perhaps be an extra thing provided by the `RunNode` type class. Then the use of the start time by the `realBlockchainTime` could become generic and work for any protocol, not just the `ProtocolRealPBFT` case.